src/engine/DnCManager.cpp

```diff
@@ -336,45 +353,42 @@ bool DnCManager::createEngines()
 
 void DnCManager::initialDivide( SubQueries &subQueries )
 {
-    const List<unsigned> inputVariables( _baseEngine->getInputVariables() );
+    auto split = std::unique_ptr<PiecewiseLinearCaseSplit>
+        ( new PiecewiseLinearCaseSplit() );
     std::unique_ptr<QueryDivider> queryDivider = nullptr;
-    if ( _divideStrategy == DivideStrategy::LargestInterval )
+    if ( _divideStrategy == DivideStrategy::Polarity )
     {
         queryDivider = std::unique_ptr<QueryDivider>
-            ( new LargestIntervalDivider( inputVariables ) );
+            ( new PolarityBasedDivider( _baseEngine ) );
     }
-    else
+    else // Default is LargestInterval
     {
-        // Default
+        const List<unsigned> inputVariables( _baseEngine->getInputVariables() );
         queryDivider = std::unique_ptr<QueryDivider>
             ( new LargestIntervalDivider( inputVariables ) );
+        QueryDivider::InputRegion initialRegion;
+        InputQuery *inputQuery = _baseEngine->getInputQuery();
+        for ( const auto &variable : inputVariables )
+        {
+            initialRegion._lowerBounds[variable] =
+                inputQuery->getLowerBounds()[variable];
+            initialRegion._upperBounds[variable] =
+                inputQuery->getUpperBounds()[variable];
+        }
+        // Add bound as equations for each input variable
+        for ( const auto &variable : inputVariables )
+        {
+            double lb = initialRegion._lowerBounds[variable];
+            double ub = initialRegion._upperBounds[variable];
+            split->storeBoundTightening( Tightening( variable, lb,
+                                                     Tightening::LB ) );
+            split->storeBoundTightening( Tightening( variable, ub,
+                                                     Tightening::UB ) );
+        }
     }
```

Comment:
The lack of symmetry between the two strategies is unclear. If it's polarity-based, we just create the divider and it takes care of everything; but if it's largest interval, there's some extra steps. Also, what is the purpose of these steps? Just to store the initial bounds in the split? Looks like we could do it in a single loop, instead of 2 loops.

Reply:
Yes, the purpose of these steps is to store the initial bounds for interval splitting. For ReLU splitting, this information is redundant.
The two loops is merged into 1.